### PR TITLE
Add network.loki.Session

### PR DESCRIPTION
**Pull request by GeopJr**
Session is a private decentralized messenger based on Signal.
`network.loki.Session.json` is based on Signal's due to Session being based on that.

### CODE REVIEW

File: network.loki.Session.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    "type": "script",
+                    "dest-filename": "session.sh",
+                    "commands": [
+                        "export TMPDIR=\"$XDG_RUNTIME_DIR/app/$FLATPAK_ID\"",
+                        "exec zypak-wrapper /app/Session/session-messenger-desktop \"$@\""
+                    ]
+                },
+                {
+                    "type": "file",
+                    "path": "network.loki.Session.appdata.xml"
```
Comment (LeandroStanger) [inline network.loki.Session.json:58]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

Comment (GeopJr) [reply network.loki.Session.json]: Should be resolved as of 9437ad913ca29a7bcb534a3cd392ade48fb95079

**Comment by barthalion** [comment]
Have you contacted upstream team about this submission? Ideally they should be the ones who maintain it.

**Comment by nedrichards** [comment]
bot, build network.loki.Session

**Comment by nedrichards** [comment]
bot, build network.loki.Session

File: network.loki.Session.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        Free and open-source: Don’t take our word for it — check Session's code yourself
+      </li>
+    </ul>
+    <p>
+      Session is free as in free speech, free as in free beer, and free of ads and
+      trackers. Session is built and maintained by the Loki Foundation, Australia’s
+      first privacy tech not-for-profit organisation. Take back your online privacy
+      today — download Session.
+    </p>
+    <p>
```
Comment (bilelmoussaoui) [inline network.loki.Session.metainfo.xml:52]: That link won't be clickable, just use the url tag to point to that page for issues

File: network.loki.Session.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>network.loki.Session.desktop</id>
```
Comment (bilelmoussaoui) [inline network.loki.Session.metainfo.xml:3]:
```suggestion
  <id>network.loki.Session</id>
  <launchable type="desktop-id">network.loki.Session.desktop</launchable>
```

File: network.loki.Session.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  <screenshots>
+    <screenshot type="default">
+      <image type="source">https://i.imgur.com/ZnHvYjo.jpg</image>
+    </screenshot>
+  </screenshots>
+  <releases>
+    <release version="1.0.8" date="2020-06-01"/>
+  </releases>
+  <content_rating type="oars-1.1">
+    <content_attribute id="violence-cartoon">none</content_attribute>
```
Comment (bilelmoussaoui) [inline network.loki.Session.metainfo.xml:66]: You can remove the ones with none as a value, it's the default one

File: network.loki.Session.json [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=pulseaudio",
+        "--share=network",
+        "--device=dri",
+        "--filesystem=home",
+        "--talk-name=org.freedesktop.Notifications",
+        "--talk-name=org.kde.StatusNotifierWatcher",
+        "--own-name=org.kde.StatusNotifierItem-2-1"
```
Comment (bilelmoussaoui) [inline network.loki.Session.json:21]: This won't really work in practice, you need `--own-name=org.kde.*` if you want a working tray icon

File: network.loki.Session.metainfo.xml (hunk 2 above)
Comment (GeopJr) [reply network.loki.Session.metainfo.xml]: Should be resolved as of 0e40e124f2332f0b245358615702301a2b019cb0

File: network.loki.Session.metainfo.xml (hunk 4 above)
Comment (GeopJr) [reply network.loki.Session.metainfo.xml]: Should be resolved as of 0e40e124f2332f0b245358615702301a2b019cb0

File: network.loki.Session.json (hunk 5 above)
Comment (GeopJr) [reply network.loki.Session.json]: Should be fixed as of 18a8112a3fbc06442357f0c8a58f0d41db0733f9

**Comment by barthalion** [comment]
Repeating my question: have you contacted upstream team about this submission? Ideally they should be the ones who at least help with maintenance. Also metainfo file should be submitted to and maintained in the upstream repo.

**Comment by GeopJr** [comment]
> Repeating my question: have you contacted upstream team about this submission? Ideally they should be the ones who at least help with maintenance. Also metainfo file should be submitted to and maintained in the upstream repo.

Yes I have! Apparently my reply to the email didn't send, sorry...
They will reply to this PR when they are ready to take over so it shouldn't be merged till then.
I'm just trying to prepare it for them.

File: network.loki.Session.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      today — download Session.
+    </p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image type="source">https://i.imgur.com/ZnHvYjo.jpg</image>
+    </screenshot>
+  </screenshots>
+  <releases>
+    <release version="1.0.8" date="2020-06-01"/>
```
Comment (Mikunj) [inline network.loki.Session.metainfo.xml:61]:
```suggestion
    <release version="1.0.9" date="2020-06-05"/>
```

File: network.loki.Session.json [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+                "install session.sh /app/bin/session",
+                "install -Dm644 network.loki.Session.metainfo.xml /app/share/metainfo/network.loki.Session.metainfo.xml"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "only-arches": [
+                        "x86_64"
+                    ],
+                    "url": "https://github.com/loki-project/session-desktop/releases/download/v1.0.8/session-messenger-desktop-linux-amd64-1.0.8.deb",
```
Comment (Mikunj) [inline network.loki.Session.json:45]:
```suggestion
                    "url": "https://github.com/loki-project/session-desktop/releases/download/v1.0.9/session-messenger-desktop-linux-amd64-1.0.9.deb",
```

**Comment by Mikunj** [comment]
> Repeating my question: have you contacted upstream team about this submission? Ideally they should be the ones who at least help with maintenance. Also metainfo file should be submitted to and maintained in the upstream repo.

Hey @barthalion, can i get more info about the statement about metainfo file in the upstream repo?

**Comment by barthalion** [comment]
@Mikunj Metainfo/appdata files essentially belong to the project code, they aren't Flathub specific. This way you can update release version/date when you create new tag and it will be also used by other distributions if they decide to package your application.

**Comment by barthalion** [comment]
bot, build network.loki.Session

**Comment by barthalion** [comment]
bot, build network.loki.Session

**Comment by barthalion** [comment]
/merge @KeeJef @Mikunj 

Let us know if you'd like to add someone else from your team should also have write access to the Flathub repo. Thanks!

**Pull request closed without merging**